wagtail/admin/views/generic/models.py

```diff
@@ -16,15 +17,65 @@ class IndexView(PermissionCheckedMixin, WagtailAdminTemplateMixin, BaseListView)
     index_url_name = None
     add_url_name = None
     edit_url_name = None
+    template_name = 'wagtailadmin/generic/index.html'
     context_object_name = None
     any_permission_required = ['add', 'change', 'delete']
+    page_kwarg = 'p'
+    default_ordering = None
+
+    def get(self, request, *args, **kwargs):
+        if not hasattr(self, 'columns'):
+            self.columns = self.get_columns()
+
+        return super().get(request, *args, **kwargs)
+
+    def get_columns(self):
+        try:
+            return self.columns
+        except AttributeError:
+            return [
```

Comment:
Any reason to not set the default value of the ``columns`` attribute to this? I can imagine this being useful for overriding as we do with similar attributes on page:

```
class MyIndexView(IndexView):
    columns = IndexView.columns + [
        # ...
    ]
```

Reply:
The problem is that we can't reference `self.get_edit_url` from a class-level definition.